src/ConnectedMode.UnitTests/UI/ManageBinding/ManageBindingViewModelTests.cs

```diff
@@ -0,0 +1,286 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel;
+using SonarLint.VisualStudio.ConnectedMode.UI.ManageBinding;
+using SonarLint.VisualStudio.ConnectedMode.UI.ProjectSelection;
+using static SonarLint.VisualStudio.ConnectedMode.ConnectionInfo;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.UI.ManageBinding;
+
+[TestClass]
+public class ManageBindingViewModelTests
+{
+    private ManageBindingViewModel testSubject;
+    private readonly SolutionInfoModel solutionInfoModel = new("VS Sample 2022", SolutionType.Solution);
+    private readonly ServerProject serverProject = new ("a-project", "A Project");
+    private readonly Connection SonarQubeConnection = new ("http://localhost:9000", ServerType.SonarQube, true);
+    private readonly Connection SonarCloudConnection = new ("http://sonarcloud.io", ServerType.SonarCloud, true);
+
+    [TestInitialize]
+    public void TestInitialize()
+    {
+        testSubject = new ManageBindingViewModel(solutionInfoModel);
+    }
+
+    [TestMethod]
+    public void IsCurrentProjectBound_ProjectIsBound_ReturnsTrue()
+    {
+        testSubject.BoundProject = serverProject;
+
+        testSubject.IsCurrentProjectBound.Should().BeTrue();
+    }
+
+    [TestMethod]
+    public void IsCurrentProjectBound_ProjectIsNotBound_ReturnsFalse()
+    {
+        testSubject.BoundProject = null;
+
+        testSubject.IsCurrentProjectBound.Should().BeFalse();
+    }
+
+    [TestMethod]
+    public void BoundProject_Set_RaisesEvents()
+    {
+        var eventHandler = Substitute.For<PropertyChangedEventHandler>();
+        testSubject.PropertyChanged += eventHandler;
+        eventHandler.ReceivedCalls().Should().BeEmpty();
+
+        testSubject.BoundProject = serverProject;
+
+        eventHandler.Received().Invoke(testSubject,
```

Comment:
There needs to be a check that he event was raised for `BoundProject` alongside this existing check

Reply:
I am not sure I understand what you mean.